CONTRIBUTING.md

```diff
@@ -232,6 +233,38 @@ please ping ([@open-telemetry/javascript-approvers](https://github.com/orgs/open
 - New or changed functionality is documented if appropriate
 - Substantial changes should not be merged within 24 hours of opening in order to allow reviewers from all time zones to have a chance to review
 
+## New Instrumentation
+
+**Do not submit pull requests for new instrumentation without reading the following.**
+
+This project is dedicated to promoting the development of quality instrumentation using OpenTelemetry.
+To achieve this goal, we recognize that the instrumentation needs to be written using the best practices of OpenTelemetry, but also by developers that understand the package they are instrumenting.
+Additionally, the produced instrumentation needs to be maintained and evolved.
+
+The size of the OpenTelemetry JavaScript developer community is not large enough to support an ever-growing amount of instrumentation.
+Therefore, to reach our goal, we have the following recommendations for where instrumentation packages should live.
+
+1. Native to the instrumented package
+2. A dedicated public repository
+3. Here in the opentelemetry-js-contrib repository
+
+If possible, OpenTelemetry instrumentation should be included in the instrumented package.
+This will ensure the instrumentation reaches all package users, and is continuously maintained by developers that understand the package.
+
+If instrumentation cannot be directly included in the package it is instrumenting, it should be hosted in a dedicated public repository owned by its maintainer(s).
```

Comment:
This is what we (the Azure SDK) team do today with our [azure sdk instrumentation package](https://www.npmjs.com/package/@azure/opentelemetry-instrumentation-azure-sdk)

What is not clear to me, and I would love to hear from you all, is how would I then onboard our instrumentation to the auto-instrumentation-node metapackage?

looking at the code it sounds trivial:
- Add @azure/opentelemetry-instrumentation-azure-sdk as a runtime dependency of auto-instrumentation-node [here](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/metapackages/auto-instrumentations-node/package.json#L51)
- Add our [Instrumentation class](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/instrumentation/opentelemetry-instrumentation-azure-sdk/src/instrumentation.ts#L22) to the [instrumentation map](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/metapackages/auto-instrumentations-node/src/utils.ts#L95)

Is that what you'd expect to see? If not, is there any documentation for onboarding to the metapackage? I'm very interested in seeing us added there and would prefer to use our existing package (sounds like y'all prefer that as well).

> Caveat is we need to export the class at the top level (today we only export a factory function), but that's an easy change to make 😄

Reply:
We'd do not plan depend on thrid-party instrumentations from an OpenTelemetry metapackage. That's mainly for security reasons (I'm not worrying about any Azure packages, but there might be third-parties requesting this treatment that we're not so confident about).  

I'm currently prototyping a way to 
- auto-load instrumentations that are present in `node_modules` as defined by an environment variable. Users would have to explicitly opt-in to the thrid-party instrumentation instead of it being loaded by default, but that's an acceptable trade-off IMO. Essentially one would be able to set something like `OTEL_NODE_LOAD_INSTRUMENTATIONS=http,grpc,npm:@azure/opentelemetry-instrumentation-azure-sdk` and that would load the http/grpc instrumentations (if present in `node_modules`) and `@azure/opentelemetry-instrumentation-azure-sdk`
- generate code for a getNodeAutoInstrumentations (or similar) function. The code-generation can run as a build step and uses the same environment variable on run-time to control which instrumentations are active. This is intended to help with bundled apps and any cases where loading from `node_modules` is not an option (that's less interesting for instrumentations than it is for propagators, resource detectors, etc as `require` is gone in bundled apps). The build step would generate code that imports the instrumentation based on a config file.

I'm not far enough along to figure out how this would work for the OTel Operator, but I'm fairly confident that we can find a solution for this as well. These tools would basically replace the functionality of the auto-instrumentation metapackage. As far as I understand, the reason that metapackage exists today is that we did not have such functionality before.

Reply:
Thanks for the information! I understand that the auto-instrumentation metapackage may not be needed in the future however it is still the recommended approach today and I'd like to get our instrumentation on there. The guidance is a bit conflicting though - I am reading that:

- OTel doesn't want to pull in 3rd party instrumentations into the OTel metapackage
- OTel doesn't want folks to add 3rd party instrumentations into this repo (that is the 3rd option in order of preference) because of the maintenance burden

I also asked this in https://github.com/open-telemetry/opentelemetry-js-contrib/issues/2238 if you prefer to engage somewhere but a set of clear next steps would be much appreciated!

If not, I can also open a PR with what I think makes sense and let y'all provide feedback that way - I'd rather know what you prefer beforehand though 

Just to be concrete: I am interested in getting the Azure SDK instrumentation into the @opentelemetry/auto-instrumentation-node metapackage. I don't particularly care which approach I need to take and am happy to follow your guidance, but am confused about which path y'all prefer.

Thanks again!